### PR TITLE
refactored animation duration to use context

### PR DESCRIPTION
**Pull request by bobular**
I think this is tidier than passing duration through multiple handlers, but let me know your thoughts

### CODE REVIEW

File: src/map/SemanticMarkers.tsx [hunk 1]
```diff
@@ -78,7 +80,7 @@ export default function SemanticMarkers({ onViewportChanged, markers, animation}
       timeoutVariable = setTimeout(
           () => {
             setConsolidatedMarkers([...markers])
-          }, animation ? animation.duration : 0
+          }, duration
```
Comment (bfox6) [inline src/map/SemanticMarkers.tsx:83]: Will getting rid of this conditional statement mean that there will be a timeout even if an animation isn't happening?

**Comment by dmfalke** [comment]
To be honest, I think this makes the code harder to follow. It also forces the consumer of `MapVEuMap` to wrap it in the context. I personally find using props to be clearer.

File: src/map/SemanticMarkers.tsx (hunk 1 above)
Comment (bobular) [reply src/map/SemanticMarkers.tsx:83]: Oh, good point.  I shouldn't have removed the conditional...

**Comment by bobular** [comment]
But won't MapVEuMap be wrapped in application config context anyway?
If so then config.map.animation.duration would be part of that.
Surely there's a need for application wide config (e.g. internationalisation).

In the meantime, I'll see if I can refactor it with props (but differently to Brett's original implementation, which for some reason rubs me up the wrong way... nothing personal Brett!)

**Comment by dmfalke** [comment]
I see what you mean. Don't we plan to release the map as a standalone
component, to be used outside the MapVEu application? If MapVEuMap is that
standalone component, then I don't think we should expect it to be wrapped
in context. We could expose a more public wrapper around MapVEuMap as the
standalone component.

In general, I think "public" components should not require context, if
possible. Hope that makes sense!

**Comment by bobular** [comment]
Yes that makes sense - thanks!

**Pull request closed without merging**